terraform/redis/cluster/provision/data.tf

```diff
@@ -75,4 +75,14 @@ data "aws_subnets" "all" {
     name   = "vpc-id"
     values = [data.aws_vpc.vpc.id]
   }
+}
+
+resource "terraform_data" "major_version" {
+  count = var.auto_minor_version_upgrade == true ? 1 : 0
```

Comment:
thoughts: I guess we could use [triggers_replace](https://developer.hashicorp.com/terraform/language/resources/terraform-data#triggers_replace) but we are already familiar with count